File: tks/ApplicationPorts/src/main/kotlin/ports/primary/ResourceManageCommandPort.java
package ports.primary;

import domain.exceptions.UnknownResourceException;
import domain.model.Resource;
import domain.model.values.AccessionNumber;

public interface ResourceManageCommandPort {

    void create(Resource resource) throws UnknownResourceException;
    void update(Resource resource);
    void remove(AccessionNumber id);
}

File: tks/ApplicationPorts/src/main/kotlin/ports/primary/ResourceManageCommandPort.kt
package ports.primary

import domain.exceptions.UnknownResourceException
import domain.model.traits.Resource
import domain.model.values.AccessionNumber

interface ResourceManageCommandPort {
    @Throws(UnknownResourceException::class)
    fun create(resource: Resource)
    fun update(resource: Resource)
    fun remove(id: AccessionNumber)
}